frontend/src/main/web/lib/components/glossary/SourceActionCell.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import {PureRenderMixin} from 'react/addons';
+import { Button, Icon, Tooltip, OverlayTrigger } from 'zanata-ui';
+import Actions from '../../actions/GlossaryActions';
+import LoadingCell from './LoadingCell'
+import DeleteEntry from './DeleteEntry'
+import GlossaryStore from '../../stores/GlossaryStore';
+import _ from 'lodash';
+
+var SourceActionCell = React.createClass({
+  propTypes: {
+    resId: React.PropTypes.string.isRequired,
+    info: React.PropTypes.string.isRequired,
+    rowIndex: React.PropTypes.number.isRequired,
+    srcLocaleId: React.PropTypes.string.isRequired,
+    canUpdateEntry: React.PropTypes.bool,
+    canDeleteEntry: React.PropTypes.bool
+  },
+
+  mixins: [PureRenderMixin],
+
+  getInitialState: function() {
+    return this._getState();
+  },
+
+  _getState: function () {
+    return {
+      entry: GlossaryStore.getEntry(this.props.resId)
+    }
+  },
+
+  componentDidMount: function() {
+    GlossaryStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    GlossaryStore.removeChangeListener(this._onChange);
+  },
```

Comment:
This pair of functions adding and removing listeners of `GlossaryStore` seems to be repeated a lot. This could probably be done with a mixin instead to reduce the duplication.

There are some other functions that are identical between several components that could use the same treatment.